app/models/ext_management_system.rb

```diff
@@ -414,6 +414,16 @@ def self.ems_infra_discovery_types
     @ems_infra_discovery_types ||= %w(virtualcenter scvmm rhevm)
   end
 
+  def disable_ems
+    # check roles\features
+    update_attributes(:enabled => false)
+  end
+
+  def enable_ems
+    # check roles\features
+    update_attributes(:enabled => true)
+  end
+
```

Comment:
contrary to #13204, I changed the attribute to `enabled` (since its a little nicer than dealing with negatives - `disabled = true`). The migration below should take care of new\existing provider by setting `enabled = true` by default.
Also, Im not sure if we should deal with `miq_product_features.yml` yet.